specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/monitors.json

```diff
@@ -0,0 +1,1190 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/monitorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProviderInstanceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProviderInstanceParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Created provider instance",
+            "schema": {
+              "$ref": "#/definitions/ProviderInstance"
+            }
+          },
+          "201": {
+            "description": "Creating provider instance",
+            "schema": {
+              "$ref": "#/definitions/ProviderInstance"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ProviderInstances"
+        ],
+        "operationId": "ProviderInstances_Delete",
+        "x-ms-long-running-operation": true,
+        "summary": "Deletes a provider instance.",
+        "description": "Deletes a provider instance for the specified subscription, resource group, Monitor name, and resource name.",
+        "x-ms-examples": {
+          "Deletes a SAP monitor provider": {
+            "$ref": "./examples/workloadmonitor/ProviderInstances_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/monitorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProviderInstanceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Deleted provider instance",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/OperationStatusResult"
+            }
+          },
+          "202": {
+            "description": "Deleting provider instance"
+          },
+          "204": {
+            "description": "No provider instance to delete"
+          },
```

Comment:
Same here

Reply:
similar explanation to monitors

Reply:
If this is being modeled as a long running operation, RPaaS should never return a 200 for this operation. So you only need a 202 and a 204 for this.

Reply:
It depends if we start a long running operation depending upon the state of the monitor if it is present or not. So we might check if skipManagedRgCleanup flag is set and if yes then return 200 to RPaaS